model/task/task.go

```diff
@@ -2484,15 +2490,52 @@ func GetTasksByVersion(versionID, sortBy string, statuses []string, variant stri
 		countPipeline = append(countPipeline, stage)
 	}
 	countPipeline = append(countPipeline, bson.M{"$count": "count"})
-	sorters := bson.D{}
+
 	if len(sortBy) > 0 {
-		sorters = append(sorters, bson.E{Key: sortBy, Value: sortDir})
+		if sortBy == DisplayStatusKey {
+			// setting field `first` onto tasks with a failed status allows us to sort all failed statuses to top of query and then sort alphabetically
+			pipeline = append(pipeline, bson.M{
+				"$addFields": bson.M{
+					"first": bson.M{
+						"$cond": bson.M{
+							"if": bson.M{
+								"$or": bson.A{
+									bson.M{"$eq": bson.A{"$" + DisplayStatusKey, evergreen.TaskFailed}},
+									bson.M{"$eq": bson.A{"$" + DisplayStatusKey, evergreen.TaskTestTimedOut}},
+									bson.M{"$eq": bson.A{"$" + DisplayStatusKey, evergreen.TaskTimedOut}},
+								},
+							},
+							"then": "a",
+							"else": "b",
+						},
+					},
+				},
+			})
+			// ordered sort with `first` at beginning of sort to sort all failure statuses to the top
+			pipeline = append(pipeline, bson.M{
+				"$sort": bson.D{
+					bson.E{Key: "first", Value: sortDir},
+					bson.E{Key: DisplayStatusKey, Value: sortDir},
+					bson.E{Key: IdKey, Value: 1},
+				},
+			})
+		} else {
+			pipeline = append(pipeline, bson.M{
+				"$sort": bson.D{
+					bson.E{Key: sortBy, Value: sortDir},
+					bson.E{Key: IdKey, Value: 1},
+				},
+			})
+		}
+	} else {
+		pipeline = append(pipeline, bson.M{
+			"$sort": bson.D{
+				// sort by _id to ensure a consistent sort order when previous sort keys result in a tie
+				bson.E{Key: IdKey, Value: 1},
```

Comment:
I think that this works if it appends to a list of sorters, but if this adds an additional sort stage, I think it would break a previously sorted list. I may be missing something though.

Reply:
this code block is only reached if no sorter is provided as a param to the function; therefore no additional sort stages will exist beside this one.

Reply:
I think this comment is a bit misleading then. Could you delete `when previous sort keys result in a tie`?